src/mame/roland/roland_d70.cpp

```diff
@@ -0,0 +1,562 @@
+// license:BSD-3-Clause
+// copyright-holders:giulioz
+/****************************************************************************
+
+    Driver for Roland D-70 synthesizer.
+
+****************************************************************************/
+
+#include "bus/generic/carts.h"
+#include "bus/generic/slot.h"
+#include "bus/midi/midiinport.h"
+#include "bus/midi/midioutport.h"
+#include "cpu/mcs96/i8x9x.h"
+#include "cpu/mcs96/i8xc196.h"
+#include "emu.h"
+#include "emupal.h"
+#include "machine/ram.h"
+#include "machine/timer.h"
+#include "screen.h"
+#include "softlist_dev.h"
+#include "sound/rolandpcm.h"
+#include "speaker.h"
+#include "video/t6963c.h"
+#include <queue>
+
+namespace {
+
+// unscramble address: ROM dump offset -> proper (descrambled) offset
+#define UNSCRAMBLE_ADDR_INT(_offset)                                           \
+  bitswap<19>(_offset, 18, 17, 15, 14, 16, 12, 11, 7, 9, 13, 10, 8, 3, 2, 1,   \
+              6, 4, 5, 0)
+// scramble address: proper offset -> ROM dump offset
+#define SCRAMBLE_ADDR_INT(_offset)                                             \
+  bitswap<19>(_offset, 18, 17, 14, 16, 15, 9, 13, 12, 8, 10, 7, 11, 3, 1, 2,   \
+              6, 5, 4, 0)
+
+// PCM cards use a different address line scrambling
+#define UNSCRAMBLE_ADDR_EXT(_offset)                                           \
+  bitswap<19>(_offset, 18, 17, 8, 9, 16, 11, 12, 7, 14, 10, 13, 15, 3, 2, 1,   \
+              6, 4, 5, 0)
+
+#define UNSCRAMBLE_DATA(_data) bitswap<8>(_data, 1, 2, 7, 3, 5, 0, 4, 6)
+
+#define CONT_MASK 0b10000000
```

Comment:
For this kind of thing we prefer static constexpr over #define.  Gives some type safety and it's more modern C++.